### PR TITLE
Add Daily PHP Tests to CI

### PR DESCRIPTION
**Pull request by adrianduffell**
Fixes (partially) #6659

Adds a daily workflow to CI, running the PHP tests against nightly builds of WordPress and WooCommerce. It uses the Docker PHP Test Suite to run the tests 🎉 

A small change was made to the `bin/install-wp-tests.sh` script to support using `latest` as a value for the `WC_VERSION` environment variable.

It was suggested in internal discussions to add Slack notifications. I'd like to see this as a follow-up effort. 

Note this reveals a failing PHP test! This should be investigated further separately.

### Detailed test instructions:

- In this PR, see the `Daily PHP Tests` checks.
- See that tests are run against latest and nightly build combinations of WordPress and WooCommerce

No changelog.

### CODE REVIEW

File: .github/workflows/daily-php.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+              with:
+                  node-version: '14'
+            - name: Build
+              run: |
+                  npm run build:feature-config	
+                  composer install  --no-dev
+              shell: bash
+            - name: Run the PHP unit tests
+              env:
+                  WP_VERSION: ${{ matrix.woocommerce }}
```
Comment (louwie17) [inline .github/workflows/daily-php.yml:33]: This should be `matrix.wordpress`.

**Review by louwie17 — CHANGES_REQUESTED** [review]
> It was suggested in internal discussions to add Slack notifications. I'd like to see this as a follow-up effort.

This would be great indeed :)

Looking at the different test runs I got slightly confused, as the ([latest, nightly)](https://github.com/woocommerce/woocommerce-admin/pull/6715/checks?check_run_id=2234174159) version appears to be running a nightly wordpress and nightly woocommerce.
The [(latest, latest)](https://github.com/woocommerce/woocommerce-admin/pull/6715/checks?check_run_id=2234174127) build seems correct.

The [(nightly, latest) ](https://github.com/woocommerce/woocommerce-admin/pull/6715/checks?check_run_id=2234174175) build appears to be running the same as the latest, latest.

The reason was because the environment variables where both being set to `matrix.woocommerce` see comment in `daily-php.yml`.

File: .github/workflows/daily-php.yml (hunk 1 above)
Comment (adrianduffell) [reply .github/workflows/daily-php.yml]: Ah, nice catch! Thank you it's now fixed in e987e13

**Comment by adrianduffell** [comment]
@louwie17 Thanks for the review. I've now fixed the environment variable issue. Could you please take another look?

**Review by louwie17 — APPROVED** [review]
@adrianduffell this looks great now, thanks for updating it 🚢 🚀

**Pull request merged**